catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactoryTest.java

```diff
@@ -1150,16 +1156,17 @@ private Filter generateTemporalFilter(JAXBElement<BinaryComparisonOpType> tempor
         QueryImpl frameworkQuery = (QueryImpl) queryFactory.getQuery(grr)
                 .getQuery();
         N temporal = null;
+        Filter queryFilter = ((QueryImpl) frameworkQuery.getFilter()).getFilter();
         if (classes.length > 1) {
-            assertThat(frameworkQuery.getFilter(), instanceOf(Or.class));
+            assertThat(queryFilter, instanceOf(Or.class));
             int i = 0;
-            for (Filter filter : ((Or) frameworkQuery.getFilter()).getChildren()) {
+            for (Filter filter : ((Or) queryFilter).getChildren()) {
                 assertThat(filter, instanceOf(classes[i++]));
                 temporal = (N) filter;
             }
         } else {
-            assertThat(frameworkQuery.getFilter(), instanceOf(classes[0]));
-            temporal = (N) frameworkQuery.getFilter();
+            assertThat(queryFilter, instanceOf(classes[0]));
```

Comment:
There should be an assertion that the length of `classes` is not empty.

Reply:
See my reply to your comment on line 1157

Reply:
[] -> :pencil2: